src/components/kubectl/port-forward.ts

```diff
@@ -234,26 +266,45 @@ function validatePortPair(portPair: string): ValidationResult {
     };
 }
 
+/**
+ * Validates if the port is a named port or withing the valid range
+ * @param validPorts List of valid named ports
+ * @param port The port to validate
+ * @param nullable Can the port be empty or zero
+ * @returns Boolean identifying if the port is valid
+ */
+function isPortValid(validPorts: ExtractedPort[], port: string, nullable?: boolean): boolean {
+    if (validPorts.map(({name}) => name).includes(port)) {
+        return true;
+    }
+    if (nullable && ['', '0'].includes(port)) {
+        return true;
+    }
+    return 0 < Number(port) && Number(port) <= MAX_PORT_COUNT;
+}
+
 /**
  * Builds and returns multiple PortMapping objects
  * @param portString A validated, user provided string containing the port mappings
+ * @param validPorts List of valid named ports
  * @returns An array containing the requested PortMappings
  */
-export function buildPortMapping(portString: string): PortMapping[] {
+export function buildPortMapping(portString: string, validPorts: ExtractedPort[] = []): PortMapping[] {
```

Comment:
Is `validPorts` the right name through all this?  It feels like other ports could be perfectly 'valid' but I think these are specifically ports that are given names in the spec.  Is that right?  If so consider renaming to `namedPorts` or `extractedPorts` or something.

On the other hand if these really are the only valid ports and anything else will be rejected then stick with the current name.